### PR TITLE
[REG-1638] Removes old state system

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Removes all of the old state system code
- Updates SDK sample bot code

- Updates agent builder patterns
see also https://github.com/Regression-Games/RegressionClient/pull/413
see also https://github.com/Regression-Games/RegressionGames/pull/458
- Updates bossroom ability bot
see also https://github.com/Regression-Games/RGBossRoom/pull/51

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Runtime/Prefabs/RGOverlayCanvas.prefab
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Prefabs/RGOverlayCanvas.prefab:1]: This is removed because there is an overlay v2 now, correct?

**Review by vontell — APPROVED** [review]
Adios, state system... it's been a good run. Hoping this simplification really helps us focus on the value areas that matter for customers. I think I just had one question, non-blocking

File: src/gg.regression.unity.bots/Runtime/Prefabs/RGOverlayCanvas.prefab
Comment (RG-nAmKcAz) [reply src/gg.regression.unity.bots/Runtime/Prefabs/RGOverlayCanvas.prefab:1]: correct, old overlay is dead/gone

**Review by abeizer — APPROVED** [review]
RIP code gen ☠️

**Review by rcornwall-reg — APPROVED** [review]
[Non Blocking]
I used the name `RGOverlayCanvasV2` to not break the original system - but now we're pulling it out the name might get confusing because there's no V1. Low priority but we could rename it to just `RGOverlayCanvas`

**Comment by vontell** [comment]
I don't get why the formatting job is running even though I disabled it in GitHub... maybe it's because the PR was created before I disabled it? 

As for the Unity Build part, I see some errors that the unity license is not valid... will look into that now

**Comment by vontell** [comment]
@RG-nAmKcAz I re-requested the build action and it worked 🤷 very weird... also I see in your new PR that the C# formatting was successfully disabled, so I think this PR is just upset because it had it originally doing that workflow.

In other words, let me know when you want to merge and I can merge it in for you with admin privileges 👍

**Pull request merged**